app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -27,7 +28,10 @@ def global_registration_vars
                    organization: (organization || User.current.default_organization || User.current.my_organizations.first),
                    location: (location || User.current.default_location || User.current.my_locations.first),
                    hostgroup: host_group,
-                   operatingsystem: operatingsystem })
+                   operatingsystem: operatingsystem,
+                   url_host: url.host,
+                   registration_url: "#{url}/register",
```

Comment:
you can use 

```ruby
register_url :host => url.host
```

to avoid appending `/register`, just in case it ever changes

similarilly you could set the `:protocol => 'https'` or based on `url.scheme`.

You can also define a new variable, registration_url_with_uuid like this

```ruby
registration_url_with_uuid = register_url(:host => url.host, :protocol => url.scheme, :uuid => 'UUID').sub('UUID', '$UUID')
```

that would avoid also appending `?uuid=xyz` in the template itself